backend/azureblob/azureblob.go

```diff
@@ -74,6 +79,24 @@ func init() {
 		Options: []fs.Option{{
 			Name: "account",
 			Help: "Storage Account Name (leave blank to use SAS URL or Emulator)",
+		}, {
+			Name: "service_principal_file",
+			Help: `Client ID and secret for use with a service principal.
+
+Leave blank normally. Needed only if you want to use a service principal instead of interactive login.
+
+First, follow the instructions at [Microsoft AD](https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals)
+to create a service principal by registering a new application. Grant this application permissions to use your storage
+container.
+
+Then, create a new client secret, and copy the application ID, client secret, and tenant ID into a JSON file, like so:
+
+	{
```

Comment:
Is this a format that the user can download? I suspect it isn't, in which case we'd normally store these in the config file. That will mean the user can copy the config file about and rclone will carry on working.

So I'd suggest 3 entries in the config file client_id/client_secret/tenant_id

I think this will simplify your logic below

Reply:
Thanks for the suggestion. I think there is a way I can simplify the instructions for creating a service principal, and as part of that auto-generate a file that users can use as a service principal file. I am going to try it out, and let me know what you think. If the UX remains awkward, I am happy to take your suggestion and create 3 additional entries in the config file.